source/FreeRTOS_TCP_Transmission_IPv4.c

```diff
@@ -298,10 +305,18 @@ void prvTCPReturnPacket_IPV4( FreeRTOS_Socket_t * pxSocket,
             }
             else
             {
-                /* Nothing to do: the buffer has been passed to DMA and will be released after use */
+                xDoRelease = pdFALSE;
+                /* The buffer has been passed to DMA and will be released after use */
             }
         } /* if( pxNetworkBuffer != NULL ) */
     } while( ipFALSE_BOOL );
+
+    if( xDoRelease == pdTRUE )
+    {
+        vReleaseNetworkBufferAndDescriptor( pxNetworkBuffer );
```

Comment:
As we are releasing the buffer here, the call to `vReleaseNetworkBufferAndDescriptor` [here](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/FreeRTOS_TCP_Transmission_IPv4.c#L171) doesnt seem to be required.